hydromt/components/spatial.py

```diff
@@ -28,27 +29,36 @@
 
 logger = getLogger(__name__)
 
-DEFAULT_REGION_FILE_PATH = "region.geojson"
 
-
-class ModelRegionComponent(ModelComponent):
+class SpatialModelComponent(ModelComponent, ABC):
     """Define the model region."""
 
+    DEFAULT_REGION_FILENAME = "region.geojson"
+    DEFAULT_HYDROGRAPHY_FILENAME = "merit_hydro"
+    DEFAULT_BASIN_INDEX_FILENAME = "merit_hydro_index"
+
     def __init__(
         self,
         model: "Model",
+        region_component: Optional[str] = None,
+        region_filename: StrPath = DEFAULT_REGION_FILENAME,
     ) -> None:
         super().__init__(model)
-        self._data: Optional[GeoDataFrame] = None
+        self.__filename: StrPath = region_filename
+        self.__data: Optional[GeoDataFrame] = None
```

Comment:
```suggestion
        self._filename: StrPath = region_filename
        self._data: Optional[GeoDataFrame] = None
```

Reply:
Oh I get it. I think the double underscore is confusing. I'd rather have the spatial component use different properties for the region data and filename else this gets confusing with the actual model data and filename. Maybe more self._region_filename and self.region_data or self._region is maybe more clear. Still not sure I like this definition of region, it conflicts with grid.region definition for example.